app.py

```diff
@@ -47,7 +49,22 @@ def create_app() -> Flask:
     health.add_check(FlaskRunningChecker())
     health.add_check(DbChecker(db))
 
-    return flask_app
+    # Configurations for Flask-Apscheduler
+    scheduler = BackgroundScheduler()
+    scheduler.add_job(
+        func=import_applications_from_queue,
+        trigger="interval",
+        seconds=flask_app.config[
+            "SQS_RECEIVE_MESSAGE_CYCLE_TIME"
+        ],  # Run the job every 'x' seconds
+    )
+    scheduler.start()
+
+    try:
+        return flask_app
+    except Exception:
+        # Shut down the scheduler when exiting the app
+        return scheduler.shutdown()
```

Comment:
when testing on Dev, `shutdown` needs to be called in order to get the scheduler trigger working. Tried to implement shutdown with `atexit.register(lambda: scheduler.shutdown())` but this doesn't have any effect. So tried the below code to get the scheduler trigger working & scheduler exists when the try block catches any exception 
```    
try:
      # To keep the main thread alive
      return app
except:
      # shutdown if app occurs except 
      scheduler.shutdown()
```

Reply:
this is one way of implementation to get scheduler running on only main thread. there are other ways which I have tried as mentioned [here ](https://stackoverflow.com/questions/14874782/apscheduler-in-flask-executes-twice)but couldn't get those working

Reply:
Why does it need to only be on the main thread? So there aren’t multiple instances?  Couldn’t we have it in its own thread or use some singleton? 

Happy with your solution if it works though!

Reply:
scheduler doesn't trigger when running on multiple instances(that's the conclusion I came into after days of testing & internet search🙂). To prevent multiple instances of scheduler it needs to run on a single thread. However this scheduler is only short-term solution for processing received messages on PAAS. We'll move to use SNS triggers once we migrate to AWS